swarm/CHANGELOG.md

```diff
@@ -4,6 +4,11 @@
 
 [PR 2857]: https://github.com/libp2p/rust-libp2p/pull/2857
 
+- Fixed panic caused by `None` being passed into `NetworkBehaviour::inject_dial_failure` on `DialError::ConnectionLimit`.
```

Comment:
Nit: The panic - if present - is in user code so technically, we are not fixing that :)

```suggestion
- Pass actual `PeerId` of dial to `NetworkBehaviour::inject_dial_failure` on `DialError::ConnectionLimit`. See [PR 2928].
```